src/controllerimpl.h

```diff
@@ -90,6 +90,11 @@ class ControllerImpl : public QObject {
   void statusUpdated(const QString& serverIpv4Gateway,
                      const QString& deviceIpv4Address, uint64_t txBytes,
                      uint64_t rxBytes);
+
+  // This signal is emitted when the Controller feature set changed.
+  // For example a feature subsystem is failing, so the feature is no longer
+  // available.
+  void controllerCapabilitiesUpdated(ControllerCapabilities capabilities);
```

Comment:
I'm not sure that I see a use case for why the feature set would change at runtime. Even if something catastrophic were to occur (like the driver crashes), it ought to be rare enough that we can just report a backend failure or something similar to let the user know that something is broken.